backend/server.py

```
@@ -1714,16 +1714,17 @@ async def github_callback(request: Request, code: str = None, state: str = None)
         )
-        await log_activity(
-            "auth", f"User {email} logged in via GitHub", user_id=user_id
-        )
 
-    # 5. Generate tokens and redirect to frontend
+    # 4. Generate tokens and redirect to frontend.
+    # log_activity is fire-and-forget — don't block the redirect on it.
+    import asyncio
+    asyncio.create_task(log_activity(
+        "auth", f"User {email} logged in via GitHub", user_id=user_id
+    ))
```

Ruff (0.15.20)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)